### PR TITLE
Added backend validation for the PC results form

### PR DESCRIPTION
**Pull request by blakelong**
Added field validation to the PCResultSerializerWrite.

### CODE REVIEW

File: indicators/serializers_new/participant_count_serializers.py [hunk 1]
```diff
@@ -123,6 +124,93 @@ class Meta:
...
+
+        Returns
+            Value of date_collected
+        """
+        program = self.context.get('program')
+
+        if program.reporting_period_start <= value <= program.reporting_period_end:
+            return value
+
+        raise exceptions.ValidationError('This date should be within the fiscal year of the reporting period.')
```
Comment (sanjuroj) [inline indicators/serializers_new/participant_count_serializers.py:147]: Any of these error messages that might be displayed by the frontend should be wrapped in translation tags and be preceded by translator comments.

**Review by sanjuroj — CHANGES_REQUESTED** [review]
Generally, nice work, really clear code.  I made a couple of line-level comments.  The other thing that you probably want to add is validation for the disaggregation error messages.  Like, direct non-duplicated value needing to be the same as the SADD disaggregation, etc... André has a good handle on those since he just had to write them for the front-end.

File: indicators/serializers_new/participant_count_serializers.py [hunk 2]
```diff
@@ -123,6 +124,93 @@ class Meta:
...
+        """
+        pattern = r"^(http(s)?|file):\/\/.+"
+        
+        if self.initial_data.get('record_name') is None or len(self.initial_data.get('record_name')) == 0:
+            raise exceptions.ValidationError('A record name must be included along with the link.')
+
+        if re.match(pattern, value) is None:
+            raise exceptions.ValidationError('Please enter a valid evidence link.')
+
+        return value
```
[the remark this answers is not in the document]
Comment (blakelong) [reply indicators/serializers_new/participant_count_serializers.py:192]: I'll add a test case just to confirm, but I believe the validate_evidence_url method won't run if the evidence_url is not present. While testing locally I was able to add a result without supplying evidence.

File: indicators/serializers_new/participant_count_serializers.py (hunk 1 above)
Comment (blakelong) [reply indicators/serializers_new/participant_count_serializers.py:147]: Sounds good, I'll get those added.

File: indicators/serializers_new/participant_count_serializers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sanjuroj) [reply indicators/serializers_new/participant_count_serializers.py:192]: Ah, gotcha, ok, if that's the case then not as much of an issue.  

However, I would definitely check out a regular result form to see if the  record name is required if there is an evidence url.  I don't believe it is.

File: indicators/serializers_new/participant_count_serializers.py (hunk 1 above)
Comment (blakelong) [reply indicators/serializers_new/participant_count_serializers.py:147]: I was looking at commits for translator comments. Would a translator comment be along the lines of `# Translators: An error message detailing that the selected date must be within the fiscal years reporting periods`

**Review by sanjuroj — APPROVED** [review]

**Pull request merged**